### ISSUE TITLE
Backup file corruption

### ISSUE THREAD

**Issue by redmop** [someone1/zfsbackup-go#1]
How does it handle the backup files being corrupted? Maybe include par as an option.

**Comment by someone1** [comment]
The original intention for this application is for the files to reside on storage targets that are not prone to such issues, such as those offered by Google, Amazon, Backblaze, Azure, etc. 

I did look into adding this feature (specifically, Reed-Solomon erasure encoding) but reasoned it was not required for this project.

If there is interest in using targets that don't have such file resiliency guarantees where adding parity bits to each file makes sense, I'd be happy to add this as an option in the future.

**Comment by redmop** [comment]
I would definitely suggest something along these lines. "file resiliency guarantees" don't help you when the file you are working with is corrupt somehow.

But then, I'm a backup nut. I use ZFS send/recv, Backuppc, Borg, and Proxmox all on the same data.

**Comment by prologic** [comment]
So S3, Google, Azure and Backblaze guarantee data integrity? -- I'm looking at using this tool/project to backup my ZFS pool to Backblaze

**Comment by someone1** [comment]
Yes, almost all intended targets come with some SLA on the durability of the data you store on there, and since zfsbackup-go supports multiple targets, you can increase your durability by utilizing multiple providers. **Note:** I am still working on adding Azure/Backblaze targets, should hit within a week or two.

Google: [99.999999999% durability](https://cloud.google.com/storage/docs/storage-classes) - they mention the usage of erasure encodings
AWS S3: [99.999999999% durability](https://aws.amazon.com/s3/faqs/#data-protection) - they mention the use of checksums on the data for integrity validation and repair (something that sounds similar to what ZFS does though I'm sure its more distributed and complicated than that)
Azure: Although no target is provided, they give an in-depth explanation of their architecture which you can read [here](http://sigops.org/sosp/sosp11/current/2011-Cascais/printable/11-calder.pdf) - they use the Reed-Solomon erasure encoding and you can increase durablitiy by increasing your redundancy options
Backblaze: [99.999999%](https://help.backblaze.com/hc/en-us/articles/218485257-B2-Resiliency-Durability-and-Availability) - they also utilize the Reed-Solomon erasure encoding and even open sourced their [code of it](https://www.backblaze.com/blog/reed-solomon/)

**Comment by prologic** [comment]
Nice! Thanks for the writeup! We should put this in the README for reference?
